### PR TITLE
Remove pygeos as required dependency

### LINKED ISSUE #12: Drop `pygeos`
Recently the bot update `0.3.0` PR ( https://github.com/conda-forge/dask-geopandas-feedstock/pull/11 ) was merged. However it did not include dependency updates. Upstream [`pygeos` was dropped]( https://github.com/geopandas/dask-geopandas/compare/v0.2.0...v0.3.0#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7L17 ). However it is still listed as a dependency here

https://github.com/conda-forge/dask-geopandas-feedstock/blob/7663128f7c8f9affb79e3e81780cafefe8560542/recipe/meta.yaml#L24

It would be good to drop the dependency in the recipe. May also be worthwhile to patch the repodata to drop the dependency. Also it may be worth enabling [Grayskull for the update bot]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#how-does-regro-cf-autotick-bot-create-automatic-version-updates ) by [updating `conda-forge.yml`]( https://conda-forge.org/docs/maintainer/conda_forge_yml.html#bot ). For example ( https://github.com/conda-forge/dask-cuda-feedstock/pull/16 )

### PR DESCRIPTION
**Pull request by weiji14**
Xref https://github.com/geopandas/dask-geopandas/pull/232, removed in dask-geopandas 0.3.0.

Also needed to add ~`distributed` to runtime requirements, and~ `fiona` to test.requires to make `pip check` work.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Based on changes from https://github.com/conda-forge/dask-geopandas-feedstock/pull/13, but without the inspect-grayskull bot addition.

Closes #12.

### CODE REVIEW

**Comment by weiji14** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml
Comment (jorisvandenbossche) [inline recipe/meta.yaml:24]: I think we still need to add an explicit `shapely >= 2`, to ensure users have the equivalent of pygeos (it really makes no sense to use dask-geopandas with older shapely, and actually even will error for certain operations)

This might mean we have to remove the pip-check for now, but I think the above is more important.

File: recipe/meta.yaml
Comment (jorisvandenbossche) [inline recipe/meta.yaml]: I am also inclined to remove this here, given that we explicitly depend on "dask-core" and not "dask", exactly to avoid pulling in all dependencies of dask in case you don't need them

File: recipe/meta.yaml
Comment (weiji14) [reply recipe/meta.yaml:24]: We can add a `run_constrained` (see https://conda-forge.org/docs/maintainer/adding_pkgs.html#constraining-packages-at-runtime), which means shapely won't be pulled in as an explicit runtime dependency, but anyone trying to install shapely alongside will have to pull in `shapely>=2`.

File: recipe/meta.yaml
Comment (weiji14) [reply recipe/meta.yaml]: But distributed is defined as required in https://github.com/geopandas/dask-geopandas/blob/v0.3.1/setup.py#L16? I actually got an error that distributed was missing when running `pip check`, which was why I added it.

File: recipe/meta.yaml
Comment (jorisvandenbossche) [reply recipe/meta.yaml:24]: For my understanding, what's the advantage of using run_constrained instead of just adding it as an explicit runtime dependency?

> but anyone trying to install shapely alongside will have to pull in shapely>=2.

That also works if shapely is already installed in the env?

Comment (jorisvandenbossche) [reply recipe/meta.yaml:24]: And to be clear, shapely is not an "optional" dependency (since the link about run_constrained mentions that as a use case). Shapely _is_ a required dependency of dask-geopandas (and also geopandas already has that dependency as well). It's just that in dask-geopandas itself we still allow you to have an older shapely if pygeos is installed (hence, the setup.py of dask-geopandas doesn't mention shapely>2 to allow this flexibility). But here if we don't add pygeos as dependency, that essentially makes shapely > 2 as a required dependency

File: recipe/meta.yaml
Comment (jorisvandenbossche) [reply recipe/meta.yaml]: Yeah, that's true. I don't really remember the discussions about adding that there and not here. Now, pypi dependencies don't have this concept of dask-core and geopandas-base. 

It might have been added for the pypi package because if you do `pip install dask` you do _not_ get distributed with it, while if you do `conda install dask` you do already get distributed as well.

File: recipe/meta.yaml
Comment (weiji14) [reply recipe/meta.yaml:24]: Since `dask-geopandas` depends on `geopandas-base` , which itself has a dependency on `shapely>=1.7.1`, there is no need to declare shapely as a runtime dependency here. I'm thinking we can leave out the `run_constrained` too then, and leave it to the user to manually install either `pygeos` or `shapely>=2.0`. Any constraint we add here on conda-forge will restrict users one way or another, and it's hard to enforce the `if not pygeos, install shapely 2.0` logic as far as I'm aware.

File: recipe/meta.yaml
Comment (weiji14) [reply recipe/meta.yaml]: Ok, removed `distributed`, let's see if the checks pass.

Comment (weiji14) [reply recipe/meta.yaml]: Yep, getting an error without `distributed` at https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=762329&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=986b1512-c876-5f92-0d81-ba851554a0a3&l=598:

```
export PREFIX=/home/conda/feedstock_root/build_artifacts/dask-geopandas_1692098287394/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p
export SRC_DIR=/home/conda/feedstock_root/build_artifacts/dask-geopandas_1692098287394/test_tmp
import: 'dask_geopandas'
import: 'dask_geopandas'
+ pip check
dask-geopandas 0.3.1 requires distributed, which is not installed.
Tests failed for dask-geopandas-0.3.1-pyhd8ed1ab_1.conda - moving package to /home/conda/feedstock_root/build_artifacts/broken
```

We could possibly patch things out here to get `pip check` to work, but ideally, the dependency specification should be fixed upstream at https://github.com/geopandas/dask-geopandas, and the conda-forge package would just follow that.

Comment (jsignell) [reply recipe/meta.yaml]: It looks like setup.py has always specified that distributed is required. So I think it is more correct to include distributed in the conda-forge recipe as well. 

It would probably be better if dask-geopandas did not depend on distributed but that feels like a separate discussion.

Comment (weiji14) [reply recipe/meta.yaml]: Ok, added `distributed` again in 7af3f6f457aba19eee3d7d85f7273a1def60c8ca. I've opened up an issue at https://github.com/geopandas/dask-geopandas/issues/258 to continue the `distributed` as a required dependency discussion.

**Review by martinfleis — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml
Comment (jorisvandenbossche) [reply recipe/meta.yaml]: It's expected that the pip check will fail, because the main thing it does is compare those dependencies, and as we have been discussing, the pip dependencies currently include `distributed`. 

To avoid dragging this PR on the distributed discussion, I pushed a small change to remove both distributed and the pip check, so we keep this PR solely focused on removing the pygeos dependency, which is useful to get in. I hope you don't mind, and thanks for the work here! It will be good to sort out the dependency discussion and make it more consistent between pip and conda.

**Comment by jorisvandenbossche** [comment]
Thanks @weiji14! 
Depending on the outcome of https://github.com/geopandas/dask-geopandas/issues/258, we can add the `pip check` section again once there is a new release of dask-geopandas, or add distributed and pip check again.

**Comment by weiji14** [comment]
Sure, I'd count removing `pygeos` as a small win already. We can always make a new build with `distributed` if needed, but let's see what happens upstream :smiley:

**Comment by jakirkham** [comment]
Thank you both! 🙏

**Pull request merged**
